react/features/base/react/components/native/PagedList.js

```diff
@@ -0,0 +1,279 @@
+
+        return (
+            <View
+                style = { [
+                    styles.pagedListContainer,
+                    disabled ? styles.pagedListContainerDisabled : null
+                ] }>
+                {
+                    pages.length > 1
+                        ? this._renderPagedList(disabled)
+                        : pages.length === 1
+                            ? React.createElement(
+
+                                // $FlowExpectedError
+                                /* type */ pages[0].component,
+                                /* props */ {
+                                    disabled,
+                                    style: styles.pagedList
+                                })
+                            : null
+                }
+            </View>
+        );
+    }
+
+    /**
+     * Constructs the style of an indicator.
+     *
+     * @param {number} indicatorIndex - The index of the indicator.
+     * @private
+     * @returns {Object}
+     */
+    _getIndicatorStyle(indicatorIndex) {
+        if (this.state.pageIndex === indicatorIndex) {
+            return styles.pageIndicatorActive;
+        }
+
+        return null;
+    }
+
+    _maybeRefreshSelectedPage: ?boolean => void;
+
+    /**
+     * Components that this PagedList displays may have a refresh function to
+     * refresh its content when displayed (or based on custom logic). This
+     * function invokes this logic if it's present.
+     *
+     * @private
+     * @param {boolean} isInteractive - If true this refresh was caused by
+     * direct user interaction, false otherwise.
+     * @returns {void}
+     */
+    _maybeRefreshSelectedPage(isInteractive: boolean = true) {
+        const selectedPage = this.props.pages[this.state.pageIndex];
+        let component;
+
+        if (selectedPage && (component = selectedPage.component)) {
+            const { refresh } = component;
+
+            typeof refresh === 'function'
+                && refresh.call(component, this.props.dispatch, isInteractive);
+        }
+    }
+
+    /**
+     * Sets the selected page.
+     *
+     * @param {number} pageIndex - The index of the selected page.
+     * @protected
+     * @returns {void}
+     */
+    _onSelectPage(pageIndex: number) {
+        return () => {
+            // eslint-disable-next-line no-param-reassign
+            pageIndex = this._validatePageIndex(pageIndex);
+
+            const { onSelectPage } = this.props;
+
+            typeof onSelectPage === 'function' && onSelectPage(pageIndex);
+
+            this.setState({ pageIndex }, this._maybeRefreshSelectedPage);
+        };
+    }
+
+    /**
+     * Renders a single page of the page list.
+     *
+     * @private
+     * @param {Object} page - The page to render.
+     * @param {boolean} disabled - Renders the page disabled.
+     * @returns {React$Node}
+     */
+    _renderPage(page, disabled) {
+        return page.component
```

Comment:
Personal opinion on style: I find using the ternary operator here counterproductive. I'd check if there is no component and return null early. Then return the View. This also reduces one level of indentation, which can help sometimes.